clisops/core/regrid.py

```diff
@@ -45,32 +62,171 @@ def wrapper_func(*args, **kwargs):
     return wrapper_func
 
 
+def check_dir(func, dr):
+    "Decorator to ensure that a directory exists."
+    if not os.path.isdir(dr):
+        os.makedirs(dr)
+
+    @functools.wraps(func)
+    def wrapper_func(*args, **kwargs):
+        return func(*args, **kwargs)
+
+    return wrapper_func
+
+
+check_weights_dir = functools.partial(check_dir, dr=weights_dir)
+
+
+# Initialize weights_dic (local weights storage)
+def weights_cache_init(weights_dir_tmp):
+    global weights_dir
+    global weights_dict_path
+    global weights_dic
+    weights_dir = weights_dir_tmp
+    CONFIG["clisops:grid_weights"]["local_weights_dir"] = weights_dir_tmp
+    if not os.path.isdir(weights_dir):
+        os.makedirs(weights_dir)
+    weights_dict_path = Path(weights_dir, "weights.json").as_posix()
+    try:
+        with open(weights_dict_path) as f:
+            weights_dic = json.load(f)
+    except FileNotFoundError:
+        weights_dic = {}
+
+
+weights_cache_init(weights_dir)
```

Comment:
This function shall initialize the local weights storage directory.  It is called once to initialize the global variable `weights_dir` as defined in roocs.ini (`CONFIG["clisops:grid_weights"]["local_weights_dir"]`), but it can be called again by the user to redefine where weights are stored to / read from. If the user calls it, the global variable `weights_dir` is overwritten as well as the CONFIG entry.

The CONFIG entry does not necessarily have to be changed, but I thought it should always hold the currently defined weight_dir instead of holding the original configuration.

I added a comment.

Reply:
Actually defining the global variable weights_dir was not working as intended, so now I rely only on the (potentially updated) CONFIG entry.